src/vehicle.cpp

```diff
@@ -2903,6 +2903,57 @@ bool CanVehicleUseStation(const Vehicle *v, const Station *st)
 	return CanVehicleUseStation(v->engine_type, st);
 }
 
+/**
+ * Get reason string why this station can't be used by the given vehicle.
+ * @param v The vehicle to test.
+ * @param st The station to test for.
+ * @return The string explaining why the vehicle cannot use the station.
+ */
+StringID GetVehicleCannotUseStationReason(const Vehicle *v, const Station *st)
+{
+	switch (v->type) {
+		case VEH_TRAIN:
+			return STR_ERROR_NO_RAIL_STATION;
+
+		case VEH_ROAD: {
+			const RoadVehicle *rv = RoadVehicle::From(v);
+			RoadStop *rs = st->GetPrimaryRoadStop(rv->IsBus() ? ROADSTOP_BUS : ROADSTOP_TRUCK);
+
+			StringID err = rv->IsBus() ? STR_ERROR_NO_BUS_STATION : STR_ERROR_NO_TRUCK_STATION;
+
+			for (; rs != nullptr; rs = rs->next) {
+				/* The vehicle is articulated and can therefore not go to a standard road stop. */
+				if (IsStandardRoadStopTile(rs->xy) && rv->HasArticulatedPart()) {
+					err = STR_ERROR_NO_STOP_ARTICULATED_VEHICLE;
+					continue;
+				}
+				/* The vehicle cannot go to this roadstop (different roadtype) */
+				if (!HasTileAnyRoadType(rs->xy, rv->compatible_roadtypes)) {
+					return RoadTypeIsRoad(rv->roadtype) ? STR_ERROR_NO_STOP_COMPATIBLE_ROAD_TYPE : STR_ERROR_NO_STOP_COMPATIBLE_TRAM_TYPE;
```

Comment:
This is broken; I do not think there should be a return here, but rather a continue like the case before. Or maybe the logic of this part of the function needs to be changed.

Start a game with trams and articulated road vehicles (e.g. generic trams + HEQS). Build a drive through tram and then add a bay station road. Build an articulated road vehicle, and create an order for that station. Result: "no compatible road type", but there is... it's just a bay.

Also, when changing this code, think about what should take precedence. If you change this to work the same as for the articulated vehicle, then a situation with an articulated bus and a bay bus and drive through tram can return either message depending on the order of the underlying road stops.